inst/schema/ReportInfo.schema.json

```diff
@@ -0,0 +1,77 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "id": "ReportInfo",
+    "type": "object",
+    "definitions": {
+        "git": {
+            "type": "object",
+            "properties": {
+                "branch": {
+                    "type": "string"
+                },
+                "ref": {
+                    "type": "string"
+                }
+            },
+            "required": [ "branch", "ref" ],
+            "additionalProperties": false
+        },
+        "error": {
+            "type": "object",
+            "properties": {
+                "message": {
+                    "type": "string"
+                },
+                "trace": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            },
+            "required": [ "message", "trace" ],
+            "additionalProperties": false
+        }
+    },
+    "properties": {
+        "id": {
+            "type": "string"
+        },
+        "name": {
+            "type": "string"
+        },
+        "success": {
+            "type": "boolean"
+        },
+        "date": {
+            "type": "string"
+        },
+        "elapsed": {
+            "type": "number"
+        },
+        "git": {
+            "oneOf": [
+                { "type": "null" },
+                { "$ref":"#/definitions/git" }
+            ]
+        },
+        "params": {
+            "oneOf": [
+                { "type": "null" },
+                { "$ref": "Parameters.schema.json" }
+            ]
+        },
+        "logfile": {
+            "type": [ "string", "null" ]
```

Comment:
is this equivalent to a oneOf?

Reply:
I think difference is `oneOf` means exactly one of but this means it can match any of them. I've updated the referenced types to allow `NULL` to make this consistent between the different properties.